x-pack/plugins/snapshot_restore/public/app/sections/home/snapshot_list/snapshot_table/snapshot_table.tsx

```diff
@@ -0,0 +1,162 @@
+          {snapshotId}
+        </EuiLink>
+      ),
+    },
+    {
+      field: 'repository',
+      name: translate('xpack.snapshotRestore.snapshotList.table.repositoryColumnTitle', {
+        defaultMessage: 'Repository',
+      }),
+      truncateText: true,
+      sortable: true,
+      // We deliberately don't link to the repository from here because the API request for populating
+      // this table takes so long, and navigating away by accident is a really poor UX.
+      render: (repository: string) => repository,
+    },
+    {
+      field: 'startTimeInMillis',
+      name: translate('xpack.snapshotRestore.snapshotList.table.startTimeColumnTitle', {
+        defaultMessage: 'Date created',
+      }),
+      truncateText: true,
+      sortable: true,
+      render: (startTimeInMillis: number) => formatDate(startTimeInMillis),
+    },
+    {
+      field: 'durationInMillis',
+      name: translate('xpack.snapshotRestore.snapshotList.table.durationColumnTitle', {
+        defaultMessage: 'Duration',
+      }),
+      truncateText: true,
+      sortable: true,
+      width: '100px',
+      render: (durationInMillis: number) => (
+        <FormattedMessage
+          id="xpack.snapshotRestore.snapshotList.table.durationColumnValueLabel"
+          defaultMessage="{seconds}s"
+          values={{ seconds: Math.round(durationInMillis / 1000) }}
+        />
+      ),
+    },
+    {
+      field: 'indices',
+      name: translate('xpack.snapshotRestore.snapshotList.table.indicesColumnTitle', {
+        defaultMessage: 'Indices',
+      }),
+      truncateText: true,
+      sortable: true,
+      width: '100px',
+      render: (indices: string[]) => indices.length,
+    },
+    {
+      field: 'shards.total',
+      name: translate('xpack.snapshotRestore.snapshotList.table.shardsColumnTitle', {
+        defaultMessage: 'Shards',
+      }),
+      truncateText: true,
+      sortable: true,
+      width: '100px',
+      render: (totalShards: number) => totalShards,
+    },
+    {
+      field: 'shards.failed',
+      name: translate('xpack.snapshotRestore.snapshotList.table.failedShardsColumnTitle', {
+        defaultMessage: 'Failed shards',
+      }),
+      truncateText: true,
+      sortable: true,
+      width: '100px',
+      render: (failedShards: number) => failedShards,
+    },
+  ];
+
+  // By default, we'll display the most recent snapshots at the top of the table.
+  const sorting = {
+    sort: {
+      field: 'startTimeInMillis',
+      direction: 'desc',
+    },
+  };
+
+  const pagination = {
+    initialPageSize: 20,
+    pageSizeOptions: [10, 20, 50],
+  };
+
+  const search = {
+    toolsRight: (
+      <EuiButton color="secondary" iconType="refresh" onClick={reload}>
+        <FormattedMessage
+          id="xpack.snapshotRestore.snapshotList.table.reloadSnapshotsButton"
+          defaultMessage="Reload"
+        />
+      </EuiButton>
+    ),
+    box: {
+      incremental: true,
+      schema: true,
+    },
+  };
```

Comment:
it would be nice to add a filter on repository name!

Reply:
Added to the to-do list!